extensions/healthie/validation/createMetricEntry.zod.ts

```diff
@@ -0,0 +1,7 @@
+import { z } from 'zod'
+
+export const fieldsValidationSchema = z.object({
+  userId: z.string().nonempty(),
+  category: z.string().nonempty(),
+  metricStat: z.string().nonempty(),
```

Comment:
I think @michal-grzelak has some custom-made Zod types for non-optional strings that you could re-use.